src/test/java/com/exactpro/th2/dataservice/healer/HealerTest.java

```diff
@@ -0,0 +1,231 @@
+/*
+ * Copyright 2020-2021 Exactpro (Exactpro Systems Limited)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exactpro.th2.dataservice.healer;
+
+import com.exactpro.cradle.CradleStorage;
+import com.exactpro.cradle.testevents.StoredTestEvent;
+import com.exactpro.cradle.testevents.StoredTestEventId;
+import com.exactpro.cradle.testevents.StoredTestEventWrapper;
+import com.exactpro.cradle.testevents.TestEventToStore;
+import com.exactpro.cradle.utils.CradleStorageException;
+import com.exactpro.th2.common.grpc.EventID;
+import com.exactpro.th2.common.grpc.EventStatus;
+import com.exactpro.th2.crawler.dataservice.grpc.CrawlerId;
+import com.exactpro.th2.crawler.dataservice.grpc.CrawlerInfo;
+import com.exactpro.th2.crawler.dataservice.grpc.DataServiceInfo;
+import com.exactpro.th2.crawler.dataservice.grpc.EventDataRequest;
+import com.exactpro.th2.crawler.dataservice.grpc.EventResponse;
+import com.exactpro.th2.crawler.dataservice.grpc.Status;
+import com.exactpro.th2.dataprovider.grpc.EventData;
+import com.exactpro.th2.dataservice.healer.cfg.HealerConfiguration;
+import com.exactpro.th2.dataservice.healer.grpc.HealerServiceImpl;
+import io.grpc.stub.StreamObserver;
+import io.grpc.testing.GrpcCleanupRule;
+import org.junit.Rule;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+@RunWith(JUnit4.class)
+public class HealerTest {
+    /**
+     * This rule manages automatic graceful shutdown for the registered server at the end of test.
+     */
+    @Rule
+    public final GrpcCleanupRule grpcCleanup = new GrpcCleanupRule();
```

Comment:
It is redundant. Or you should use the internal channel and test the handler using the gRPC stub